quickstart/201-k8s-cluster-with-tf-and-aks/sp.tf

```diff
@@ -0,0 +1,28 @@
+# Create Azure AD App Registration
+resource "azuread_application" "app" {
+  display_name = "my-app"
+  owners       = [local.current_user_id]
+}
+
+# Create Service Principal
+resource "azuread_service_principal" "app" {
+  application_id               = azuread_application.app.application_id
+  app_role_assignment_required = true
+  owners                       = [local.current_user_id]
+}
+
+# Create Service Principal password
+resource "azuread_service_principal_password" "app" {
+  service_principal_id = azuread_service_principal.app.id
+}
+
```

Comment:
We have a cache issue here, when we tried to create the aks cluster right after the creation of azure ad service principal and password, the aks cluster's creation would fail and complained "cannot find the application". I guess there's a cache issue on the service side so I tried the following `time_sleep` resource, and after 30s sleep the apply succussed.

```hcl

resource "time_sleep" "wait_30_seconds" {
  create_duration = "30s"

  depends_on = [azuread_service_principal_password.app]
}
```